engine.py

```diff
@@ -554,6 +554,9 @@ def label(cls, model, field=None, lang=None):
         if not lang:
             lang = Transaction().language
 
+        if not model:
```

Comment:
Why is this necessary. Why would anyone ask for a label of an empty model?

The only possibility I see is that the model may not exist and thus line 562 would crash:

model, = Model.search([('model', '=', model)])

Then we could return model variable instead of model.name.

Reply:
... crash when try to get label in a model that not exist (header report) because "model" is None.

I don't remember exact the label that crash.... :( maybe is related to signature template [1] ? 

Example:

{{ label(None, 'reference') }}

[1] https://github.com/trytonspain/trytond-html_report/blob/master/html.py#L140